Add support for caching multipart in writethrough mode (#13507)
Minio-io/minio: docs/disk-caching/DESIGN.md

  Note that cache KMS master key is not recommended for use in production deployments. If the MinIO server/gateway machine is ever compromised, the cache KMS master key must also be treated as compromised.
  Support for external KMS to manage cache KMS keys is on the roadmap,and would be ideal for production use cases.

- `MINIO_CACHE_COMMIT` setting of `writethrough` allows caching of multipart uploads synchronously if enabled. By default, single PUT operations are already cached on write without any special setting.

> Note that `MINIO_CACHE_COMMIT` also has a setting of `writeback` which allows staging single uploads in cache before committing to remote. However for consistency reasons, staging uploads in cache is not permitted for multipart uploads.Partially cached stale uploads older than 24 hours are automatically cleaned up.

Reviewer: ```suggestion
NOTE:  `MINIO_CACHE_COMMIT` also has value of `writeback` which allows staging single uploads in cache before committing to remote. However, for consistency reasons, staging uploads in the cache are not permitted for multipart uploads. Partially cached stale uploads older than 24 hours are automatically cleaned up.
```

> 

Reviewer: shouldn't this be more meaningful with `MINIO_CACHE_COMMIT_MODE` ?   .. I see _CACHE_COMMIT is already present how about we change that to _MODE and deprecate just _COMMIT
